sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/utils/HadoopInputFormatUtils.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.hadoop.inputformat.utils;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+public class HadoopInputFormatUtils {
+	private static final Set<Class> immutableTypes = new HashSet<Class>(
+			Arrays.asList(String.class, Byte.class, Short.class, Integer.class,
+					Long.class, Float.class, Double.class, Boolean.class,
+					BigInteger.class, BigDecimal.class));
+
+	public static boolean isImmutable(Object o) {
```

Comment:
Done. Earlier this was in IO file, but then we discussed of moving it out, hence added the Utils file. Now this is back to IO class, and removed Utils